File: api/Theta.Core/Entities/ItineraryAggregate/Activity.cs
public class Activity : BaseEntity
{
    public string Name { get; set; } = null!;
    public DateTime? StartTime { get; set; }
    public DateTime? EndTime { get; set; }
    public Place? Place { get; set; }
    public IEnumerable<string> Tags { get; set; } = new List<string>();
}

File: project/theta_app/src/app/feature/itinerary/models/activity.ts
import { BaseModel } from '../../../shared/models/base-model'
import { Place } from '../../../shared/models/place';
import { Image } from '../../../shared/models/image'

export interface Activity extends BaseModel {
    id: number;
    activity: string;
    day: number;
    startTime: string;
    endTime: string;
    tags: string;
    price: number;
    description: string;
    sequence: number;
    place: Place;
    images: Image[]
}